src/map.c

```diff
@@ -232,7 +232,7 @@ void resize_map(
     uint32_t bucket_count)
 {
     uint32_t *old_buckets = map->buckets;
-    uint32_t old_bucket_count = map->bucket_count;
+    uint32_t old_bucket_count = (uint32_t)map->bucket_count;
```

Comment:
I would make the `old_bucket_count` variable of type `size_t` so that you don't need a cast.